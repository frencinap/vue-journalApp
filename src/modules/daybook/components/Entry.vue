<template>
  <div class="entry-container mb-3 pointer p-2"
  @click="$router.push({name: 'entry', params: {id:10}})">
      <!-- Titulo -->
      <div class="entry-title d-flex">
          <span class="text-success fs-5 fw-bold">19</span>
          <span class="mx-1 fs-5">Noviembre</span>
          <span class="mx-2 fw-light">2021, Viernes</span>
      </div>
      <div class="entry-description">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Ex earum dicta iste aspernatur, culpa saepe excepturi repellat vitae, voluptatum reiciendis ab quidem laudantium nemo mollitia. Ipsa saepe aliquam necessitatibus deleniti!
      </div>
  </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container{
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover{
    background-color: lighten($color: grey, $amount: 45);
  }
  .entry-description{
  font-size: 12px;
}
}

</style>