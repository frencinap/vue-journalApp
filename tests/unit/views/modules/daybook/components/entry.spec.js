import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
//como necesitamos entry usamos el mock-store
import { journalState } from '../../../mock-data/test-journal-state'

describe('Pruebas en entry Component', () => {
    //mockrouter
    const mockRouter = {
        push: jest.fn()
    }
    //wrapper
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('Debe hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    });

    test('Debe redireccionar al hacer click en el entry-container', () => {
        
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        })
    });

    test('pruebas en las propiedades computadas', () => {
        // console.log(wrapper.vm.day)
        // console.log(wrapper.vm.month)
        // console.log(wrapper.vm.yearDay);
        expect( wrapper.vm.day).toBe(29)
        expect( wrapper.vm.month).toBe('Noviembre')
        expect( wrapper.vm.yearDay).toBe('2021, Lunes')
    });

})