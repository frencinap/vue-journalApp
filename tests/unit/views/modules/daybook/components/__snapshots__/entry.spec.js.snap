// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en entry Component Debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">29</span><span class="mx-1 fs-5">Noviembre</span><span class="mx-2 fw-light">2021, Lunes</span></div>
  <div class="entry-description">Estoy aprendiendo Vuejs con el profesor Fernando Herrera</div>
</div>
`;
